pkg/operator/ceph/cluster/cluster.go

```diff
@@ -230,12 +234,84 @@ func (c *ClusterController) initializeCluster(cluster *cluster) error {
 	cluster.mons.ClusterInfo = cluster.ClusterInfo
 	cluster.mons.ClusterInfo.SetName(c.namespacedName.Name)
 
+	// Skip the new CSI-operator creation when holder pod is enabled until multus support is added in the CSI operator
+	if csi.EnableCSIOperator && !csi.IsHolderEnabled() {
+		err = c.createCSIOpCephCluster(cluster.ClusterInfo, cluster)
+		if err != nil {
+			return errors.Wrap(err, "failed to configure csi operator")
+		}
+	}
+
 	// Start the monitoring if not already started
 	c.configureCephMonitoring(cluster, cluster.ClusterInfo)
 
 	return nil
 }
 
+func (c *ClusterController) createCSIOpCephCluster(clusterInfo *cephclient.ClusterInfo, cluster *cluster) error {
+
+	logger.Info("Creating ceph-CSI operator ceph cluster CR")
+	csiCephCluster := &csiopv1a1.CephCluster{}
+
+	csiCephCluster.Name = cluster.namespacedName.Name
+	csiCephCluster.Namespace = cluster.namespacedName.Namespace
+
+	err := c.createCSIOpCephClusterSpec(clusterInfo, cluster, &csiCephCluster.Spec)
+	if err != nil {
+		return errors.Wrap(err, "failed to set csi-operator operator spec")
+	}
+
+	err = c.client.Create(c.OpManagerCtx, csiCephCluster)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "failed to create CSI-operator ceph cluster CR %s", csiCephCluster.Name)
+		}
+		err = c.client.Get(c.OpManagerCtx, types.NamespacedName{Name: csiCephCluster.Name, Namespace: csiCephCluster.Namespace}, csiCephCluster)
+		if err != nil {
+			return errors.Wrap(err, "failed to get csi-operator ceph cluster CR")
+		}
+		err := c.createCSIOpCephClusterSpec(clusterInfo, cluster, &csiCephCluster.Spec)
+		if err != nil {
+			return errors.Wrap(err, "failed to update csi-operator operator spec")
+		}
+		err = c.client.Update(c.OpManagerCtx, csiCephCluster)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update CSI-operator ceph cluster CR %s", csiCephCluster.Name)
+		}
+	}
+
+	logger.Info("Successfully created ceph-CSI operator ceph cluster CR")
+	return nil
+}
+
+func (c *ClusterController) createCSIOpCephClusterSpec(clusterInfo *cephclient.ClusterInfo, cluster *cluster, createCSIOpCephClusterSpec *csiopv1a1.CephClusterSpec) error {
+	if cluster.Spec.CSI.ReadAffinity.Enabled {
+		createCSIOpCephClusterSpec.ReadAffinity.CrushLocationLabels = cluster.Spec.CSI.ReadAffinity.CrushLocationLabels
+	}
+
+	cephRBDMirrorList := &cephv1.CephRBDMirrorList{}
+	err := c.client.List(c.OpManagerCtx, cephRBDMirrorList, &client.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(cephRBDMirrorList.Items) == 0 {
+		logger.Debug("no ceph subvolumegroup found")
+	} else {
+		if cephRBDMirrorList.Items[0].DeletionTimestamp.IsZero() {
+			createCSIOpCephClusterSpec.RbdMirrorDaemonCount = cephRBDMirrorList.Items[0].Spec.Count
+		}
```

Comment:
@travisn is this right place to list and update the CSI-operator cephCLuster cr field `RbdMirrorDaemonCount`?

Reply:
@Madhu-1 We had some discussion about this rbd mirror daemon count, but I can't find the notes. What exactly is needed for that value? As I recall we don't actually need to query the cephrbdmirror CRs.